### PR TITLE
Feat server

### PR DESCRIPTION
**Pull request by DeimosA**

### CODE REVIEW

File: core/src/no/group15/playmagic/PlayMagic.kt [hunk 1]
```diff
@@ -1,28 +1,24 @@
...
-			field = value
-			appState?.create()
-			appState?.resize(Gdx.graphics.width, Gdx.graphics.height)
-		}
+	private lateinit var appState: AppState
+	private lateinit var injectContext: Context
 
 	private val commonInput = object : InputAdapter() {
 		override fun keyUp(keycode: Int): Boolean {
 			if (keycode == Input.Keys.ESCAPE || keycode == Input.Keys.BACK) {
```
Comment (skykanin) [inline core/src/no/group15/playmagic/PlayMagic.kt]: Can be rewritten to `if (keycode in setOf(Input.Keys.ESCAPE, Input.Keys.BACK) ...`

File: core/src/no/group15/playmagic/ui/views/menulists/MainMenuList.kt [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			alignment,
+			hoverBackground
+		) {
+			override fun click(x: Float, y: Float) {
+				mainMenu.startGame(NetworkContext(
+					ClientConfig(host = "localhost"),
+					ServerConfig(host = "localhost", maxPlayers = 1)
+				))
+			}
+		})
```
Comment (skykanin) [inline core/src/no/group15/playmagic/ui/views/menulists/MainMenuList.kt]: This formatting looks a bit weird to me, but if you ran the formatter in android studio and this is the result then it's fine

**Review by skykanin — CHANGES_REQUESTED** [review]
Just a couple of pedantic comments, otherwise looks good

File: core/src/no/group15/playmagic/ui/views/menulists/MainMenuList.kt (hunk 2 above)
Comment (DeimosA) [reply core/src/no/group15/playmagic/ui/views/menulists/MainMenuList.kt]: No, the autoformatter puts NetworkContext on a new line, I just thought it looked cleaner without it.

File: core/src/no/group15/playmagic/PlayMagic.kt (hunk 1 above)
Comment (DeimosA) [reply core/src/no/group15/playmagic/PlayMagic.kt]: 👍

File: core/src/no/group15/playmagic/PlayMagic.kt [hunk 3]
```diff
@@ -10,16 +10,31 @@ import no.group15.playmagic.ui.AppState
...
 
 	private lateinit var appState: AppState
 	private lateinit var injectContext: Context
 
+	// Application wide input commands
 	private val commonInput = object : InputAdapter() {
 		override fun keyUp(keycode: Int): Boolean {
-			if (keycode == Input.Keys.ESCAPE || keycode == Input.Keys.BACK) {
+			// Uses the back button on Android or escape key to navigate "back" with the state defining what back means
+			if (keycode in setOf(Input.Keys.ESCAPE, keycode == Input.Keys.BACK)) {
```
Comment (skykanin) [inline core/src/no/group15/playmagic/PlayMagic.kt]: Shouldn't it just be `setOf(Input.Keys.ESCAPE, Input.Keys.BACK)`?

**Review by skykanin — CHANGES_REQUESTED** [review]

File: core/src/no/group15/playmagic/PlayMagic.kt (hunk 3 above)
Comment (DeimosA) [reply core/src/no/group15/playmagic/PlayMagic.kt]: oh, bugger, fixed

**Review by skykanin — APPROVED** [review]
Looks good

**Pull request merged**